src/server/pfs/db/index.go

```diff
@@ -0,0 +1,220 @@
+package persist
+
+import (
+	"github.com/pachyderm/pachyderm/src/server/pfs/db/persist"
+
+	"github.com/dancannon/gorethink"
+)
+
+var (
+	DiffPathIndex    = NewDiffPathIndex()
+	DiffPrefixIndex  = NewDiffPrefixIndex()
+	DiffParentIndex  = NewDiffParentIndex()
+	DiffClockIndex   = NewDiffClockIndex()
+	ClockBranchIndex = NewClockBranchIndex()
+	CommitClockIndex = NewCommitClockIndex()
+
+	// Collect them all for easier initialization
+	Indexes = []Index{
+		DiffPathIndex,
+		DiffPrefixIndex,
+		DiffParentIndex,
+		DiffClockIndex,
+		ClockBranchIndex,
+		CommitClockIndex,
+	}
+)
+
+// An Index is a rethinkdb index.
+type Index interface {
+	GetName() string
+	GetTable() Table
+	GetCreateFunction() func(gorethink.Term) interface{}
+	GetCreateOptions() gorethink.IndexCreateOpts
+}
```

Comment:
What's the point in obscuring this behind an interface and having getter methods instead of just make `index` public and having the functions below return one directly?